src/posix/platform/udp.cpp

```diff
@@ -257,9 +255,9 @@ otError otPlatUdpBind(otUdpSocket *aUdpSocket)
     otError error = OT_ERROR_NONE;
     int     fd;
 
-    assert(sPlatNetifIndex != 0);
+    assert(gNetifIndex != 0);
     assert(aUdpSocket->mHandle != nullptr);
-    VerifyOrExit(sPlatNetifIndex != 0, error = OT_ERROR_INVALID_STATE);
+    VerifyOrExit(gNetifIndex != 0, error = OT_ERROR_INVALID_STATE);
```

Comment:
duplicate check for `gNetifIndex != 0` ?

Reply:
assert() becomes empty in release build.

Reply:
But it's very common to use OT_ASSERT without addition VerifyOrExit in OpenThread core code.

assert already means it should not happen unless there is a logical error? Do we need to use `VerifyOrExit` for logically error conditions?